src/python/turicreate/toolkits/sound_classifier/sound_classifier.py

```diff
@@ -825,11 +788,8 @@ class as a vector. Label ordering is dictated by the ``classes``
             if batch.pad != 0:
                 batch_data = batch_data[:-batch.pad]    # prevent batches looping back
 
-            batch_data = mx.gluon.utils.split_and_load(batch_data, ctx_list=ctx, batch_axis=0, even_split=False)
-
-            for x in batch_data:
-                forward_output = self._custom_classifier.forward(x)
-                y += mx.nd.softmax(forward_output).asnumpy().tolist()
+            forward_output = self._custom_classifier.predict(batch_data)
+            y += mx.nd.softmax(mx.nd.array(forward_output[0])).asnumpy().tolist()
```

Comment:
This leaves some MXNet code outside of the MXNet encapsulation. Could we move the application of `softmax` inside `predict`?